downloads/swan-lake-release-notes/swan-lake-2201.5.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,498 @@
+
+### Improvements
+
+#### Support for command-line arguments of built-in subtypes
+
+The command-line arguments of the Ballerina built-in subtypes are now supported to be parsed as operands.
+
+For the following `main` function,
+```ballerina
+public function main(byte byteVal, string:Char charVal, int:Signed8 int8Val) {
+}
+```
+
+the values can be passed through command-line arguments as follows.
+
+```
+$ bal run -- 1 b 33
+```
+
+#### Support ambiguous union-type configurable variables
+
+When a structural value is provided for a configurable variable of a union type that includes more than one type descriptor, the inherent type used will be the first (leftmost) type descriptor from which the value can be constructed.
+
+For example, in the following Ballerina code,
+```ballerina
+type Person record {|
+    string name;
+    string city;
+|};
+
+configurable map<string>|Person configVar = ?;
+```
+and when the `Config.toml` file contents are as follows,
+```toml
+configVar = {name = "Jack", city = "Colombo"}
+```
+`configVar` is instantiated with the `map<string>` inherent type.
+
+#### Support binding of resource functions to a generic native method
+
+A new way has been introduced to support the binding of any resource method to a generic native method regardless of the resource path parameters. The generic native method should be defined with a `BArray` parameter, which represents all the path parameters. To avoid errors due to overloaded methods, it is recommended to define the parameter type constraints as well.
+
+For example, the following Ballerina resource method,
+
+```ballerina
+isolated resource function get abc/[int p1]/[string p2]/[string p3]/[int... p4](string s) = @java:Method {
+    'class: "javalibs.app.App",
+    name: "getResource",
+    paramTypes: ["io.ballerina.runtime.api.values.BObject", "io.ballerina.runtime.api.values.BArray", "io.ballerina.runtime.api.values.BString"]
+} external;
+```
+
+can be bound to the following Java method.
+
+```java
+public static void getResource(BObject client, BArray path, BString str) {
+}
+```
+
+#### Improvements in Runtime Java APIs
+
+- Deprecated the `getType` runtime API, which returns an `ObjectType` in the `io.ballerina.runtime.api.values.BObject` class. Instead, a new `getOriginalType` API, which returns the `Type` is introduced to return both the `ObjectType` and the type-reference type.
+- Deprecated the `XMLNS` Java constant in the `io.ballerina.runtime.api.values.BXmlItem` runtime class. Instead, the `javax.xml.XMLConstants.XMLNS_ATTRIBUTE` constant needs to be used.
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.5.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.5.0+label%3ATeam%2FjBallerina+label%3AType%2FBug+is%3Aclosed).
+
+## Standard library updates
+
+### New features
+
+#### `http` package
+- Introduced a new HTTP status code error structure.
```

Comment:
```suggestion

- Introduced a new HTTP status code error structure.
```

Reply:
Some other places that aren't consistent too.

Reply:
New lines will get fixed automatically when the web pages are rendered.